library/mr_provisioner_get_ip.py

```diff
@@ -0,0 +1,193 @@
+    for it. It should be noted that the current behaviour is of fetching the
+    reserved address for an dynamic-reserved interface. If it's static or
+    dynamic, the module will fetch the last address given by KEA"
+
+options:
+    mrp_url:
+        description:
+            - This is the URL of the Mr Provisioner
+        required: true
+    mrp_token:
+        description:
+            - This is the authentication token for MrP's API
+        required: true
+    machine_name:
+        description:
+            - This is the machine name as shown in MrP
+        required: true
+    interface_name:
+        description:
+            - This is the name of the machine's interface you'd like the IP of.
+        required: true
+
+author:
+    - Baptiste Gerondeau (baptiste.gerondeau@linaro.org)
+'''
+
+EXAMPLES = '''
+- name: Get IP of provisioned machine
+  mr_provisioner_get_ip:
+    mrp_url: "{{ mr_provisioner_url }}"
+    mrp_token: "{{ mr_provisioner_auth_token }}"
+    machine_name: "{{ mr_provisioner_machine_name }}"
+    interface_name: "{{ mr_provisioner_interface_name|default('eth1') }}"
+  register: get_ip
+- debug: var=get_ip
+
+#Note that you get the ip fetched via get_ip['ip'] : you NEED to register get_ip
+#This seems overly complicated but I've found no other way to do it...
+'''
+
+RETURN = '''
+ip:
+    description: An.... IP !! (v4 because MrP doesn't do v6)
+    type: str
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+class ProvisionerError(Exception):
+    def __init__(self, message):
+        super(ProvisionerError, self).__init__(message)
+
+class IPGetter(object):
+    def __init__(self, mrpurl, mrptoken, machine_id, interface_name =
+                 'eth1'):
+        self.mrp_url = mrpurl
+        self.mrp_token = mrptoken
+        self.interface = interface_name
+        self.machine_id = machine_id
+        self.machine_ip = ''
+
+    def get_interfaces(self):
+        headers = {'Authorization': self.mrp_token}
+        url = urljoin(self.mrp_url,
+                      "/api/v1/machine/{}/interface".format(self.machine_id))
+        r = requests.get(url, headers=headers)
+        if r.status_code != 200:
+            raise ProvisionerError('Error fetching {}, HTTP {} {}'.format(self.mrp_url, r.status_code,
+                                                         r.reason))
+        if len(r.json()) == 0:
+            raise ProvisionerError('Error no machine with id "{}"'.format(self.machine_id))
+
+        return r.json()
+
+    def get_ip(self):
+        try:
+            interfaces = self.get_interfaces()
+        except ProvisionerError as e:
+            print('Could not fetch interface for machine : "{}"'.format(e))
+            return 'FAILURE'
+
+        for i in interfaces:
+            if str(i['identifier']) == self.interface:
+                if str(i['config_type_v4']) == 'dynamic-reserved' and str(i['configured_ipv4']):
```

Comment:
I think we can simply return 'lease_ipv4'. When the interface configured as 'dynamic-reserved', 'lease_ipv4' equals to 'configured_ipv4'